static/js/controllers/edit-contact.js

```diff
@@ -53,7 +53,13 @@ var libphonenumber = require('libphonenumber/utils'),
 
       populateParents();
 
-      $scope.$on('ContactUpdated', populateParents);
+      $scope.$on('ContactUpdated', function(e, contact) {
+        if (contact && contact._deleted) {
+          // allow the ContactCtrl to handle contact deletion
+          return;
+        }
+        populateParents();
+      });
```

Comment:
This will leave the EditContact modal in an inconsistent state - ie: it will still have old deleted contacts in the parent dropdown.

Because the EditContact modal is in the root context I think it should be allowed to complete even if navigation occurs, so see if you can set `timeout:false` on this request.